Major Core Refactor for UI

lental opened the pull request:
Split OoyalaSkinCore into OoyalaSkinPlayerListener and OoyalaSkinPanelRenderer

Made Overlays part of the Render code

Keep and OverlayStack, which lets you pop and push overlays onto it

Put Panels into a separate folder.

Cleaned up a lot of panels styles

lental committed:
Major Overlay Refactor - first order overlays

Instead of More Options Screen rendering Discovery and Closed Captions, pull them out and render them separately

Maintain a stack of overlays so we can go back when we need to

lental committed:
move All SkinCore Listener code to a separate file

lental committed:
Pull all render code from ooyalaSkinCore into ooyalaSkinPanelRenderer

abstraction and isolation

lental committed:
create ads panel

lental committed:
Move all panels to a panels folder, delete unused panels

lental committed:
properly quote a json style

lental committed:
Support Share from Control Bar (does not pause), cleanup pause code

previousScreentype is no longer necessary because of the Overlay Stack

lental committed:
Rename panel methods to avoid the "on*()" style.

For consistency's sake, on*() is used for listening to notification/events, handle is for button presses and interactions

lental commented:
I will merge this in before the candidate RC creation on Sunday.  If you can, please review.

Thanks,
Mike

Matthew-C commented:
I will take a look.
On Mar 12, 2016 1:36 PM, "lental" notifications@github.com wrote:

> I will merge this in before the candidate RC creation on Sunday. If you
> can, please review.
> 
> Thanks,
> Mike
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ooyala/ios-skin/pull/383#issuecomment-195811931.

Matthew-C commented on sdk/react/index.android.js:
```diff
@@ -89,34 +90,15 @@ var OoyalaSkin = React.createClass({
-      default: 
-        return OoyalaSkinCoreInstance.renderVideoView();     
-         break;
+    Log.verbose("Rendering - Current Overlay stack: " + this.state.overlayStack);
```
both iOS and android have the same logic, shall we move it to ooyalaCore?

lental committed:
Fix Closed Captions in Android, separate Ad View to its own view

I know there's duplicate VideoView code, but we can refactor that soon

Removed ad code in VideoView

Handle touch of controls to avoid the autohide while handling controls

Fixed discovery end screen by modifying state after suspend

lental committed:
Move Duplicate render code from indexes to skinCore

Matthew-C committed:
Merge pull request #384 from ooyala/additionalCoreRefactor

Additional core refactor
